app/controllers/api/v1/recordings_controller.rb

```diff
@@ -70,6 +70,11 @@ def update_visibility
         render_data status: :ok
       end
 
+      def recordings_count
+        count = current_user.recordings.count
```

Comment:
Can't we just include this to the `SessionsController#index` API and simply extract it fromt the `AuthContext`?

Reply:
However, I think we should test this endpoint?

Reply:
> Can't we just include this to the `SessionsController#index` API and simply extract it fromt the `AuthContext`?

IMO this does not belong there. And we would need to "drill down" the value.

Reply:
The `AuthContext` is a context, what you mean by drill down?
Contexts were introduced to resolve the "props drilling" issue if that's what you're mentioning,
My idea is that we already have queries that are sent to fetch the current user state,
Anything like the current user name, email, active status, ... along with its permissions and their values can be collected in that API call,
That state is provided to the client application through a context where any component no matter where in the subtree can subscribe to and extract the current user data,
My idea was why making a separate endpoint when we can leverage the  infrastructure that we have through `AuthContext` and via the `SessionsController#index` API.

Reply:
I think it's the same of what we've done with the `CurrentUserSerializer` for the current user permissions,

Reply:
I'm not sure what Contexts are doing TBH.
What's the difference between calling useAuth() or useRecordingsCount() to get our value?

Reply:
The only difference I see is the memoization in AuthContext? So that the recording count would be fetch from memory in your solution? Instead of a query.

Reply:
An API call (minimum 1) + an extra entry in our queryCache to hold the data + any work involved from react-query to manage that cache.

Reply:
However, to be aware  if you're willing to leverage the `SessionsController#index` means that any mutation to the recordings should invoke an invalidation to the `useSessions` cache which may cause an overhead,

Reply:
Am I correct in my assumption above? 

About the useAuth() not firing an extra API call because of the memoization? And that memoization is the core of a Context?

Or the added value of your solution is mostly in the queryCache/React Query business?